database/ApplicationDbContext.cs

```diff
@@ -62,6 +62,17 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 .Entity<ScTrialBookingRequest>()
                 .HasIndex(r => new { r.CaseNumber }, "IX_CaseNumber_Ascending");
 
+            modelBuilder
+                .Entity<ScTrialBookingRequest>()
+                .HasIndex(
+                    r => new { r.Email, r.RequestedByName },
```

Comment:
Include the ProcessingTimestamp here too as the first column in the index.   Entiry framework is querying 3 columns in your query. 
```
.Where(e => e.ProcessingTimestamp < oldestDate)
.Where(e => e.Email != anonymizedEmail && e.RequestedByName != anonymizedName)
```

Reply:
I didn't know that's how it works! Kind of a bummer that it won't benefit from multiple indexes, but that's great to know